centreon/src/Core/ResourceAccess/Infrastructure/Repository/DbWriteResourceAccessRepository.php

```diff
@@ -49,6 +50,87 @@ public function __construct(DatabaseConnection $db, \Traversable $repositoryProv
         $this->repositoryProviders = iterator_to_array($repositoryProviders);
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function deleteDatasets(array $ids): void
+    {
+        $fields = '';
+        foreach ($ids as $index => $id) {
+            $fields .= ('' === $fields ? '' : ', ') . ':id_' . $index;
+        }
+
+        $request = <<<SQL
+                DELETE FROM `:db`.acl_resources WHERE acl_res_id IN ({$fields})
+            SQL;
+
+        $statement = $this->db->prepare($this->translateDbName($request));
+
+        foreach ($ids as $index => $id) {
+            $statement->bindValue(':id_' . $index, $id, \PDO::PARAM_INT);
+        }
+
+        $statement->setFetchMode(\PDO::FETCH_ASSOC);
+        $statement->execute();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function update(Rule $rule): void
+    {
+        $alreadyInTransaction = $this->db->inTransaction();
+        if (! $alreadyInTransaction) {
+            $this->db->beginTransaction();
+        }
+
+        try {
+            $this->updateBasicInformation($rule);
+
+            if (! $alreadyInTransaction) {
+                $this->db->commit();
+            }
```

Comment:
the transaction mechanism is also managed in the useCase

```
$this->dataStorageEngine->startTransaction();
            $this->updateBasicInformation($rule, $request);
```

The method updateBasicInformation call the `update` method

Reply:
> the transaction mechanism is also managed in the useCase
> 
> ```
> $this->dataStorageEngine->startTransaction();
>             $this->updateBasicInformation($rule, $request);
> ```
> 
> The method updateBasicInformation call the `update` method

yes, but the transaction levels are not the same